src/components/TwoPaneView/TwoPaneView.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import bem from 'bem';
+import { BlockWidthRestrictor } from '../../index';
+import LeftPane from './LeftPane';
+import RightPane from './RightPane';
+import styles from './TwoPaneView.scss';
+
+const isNotPane = element => element && element.type !== RightPane && element.type !== LeftPane;
+
+const { block } = bem({
+    name: 'TwoPaneView',
+    classnames: styles,
+});
+
+const TwoPaneView = props => {
+    const { children, ...rest } = props;
+
+    const [rightTop, setRightTop] = React.useState(0);
+    const [rightHeight, setRightHeight] = React.useState(200);
+    const [rightWidth, setRightWidth] = React.useState(600);
+    const [doDisplayRightPane, setDoDisplayRightPane] = React.useState(true);
+
+    const leftRef = React.createRef();
+    const blockRef = React.createRef();
+
+    const handleScroll = (e, leftEl = leftRef.current) => {
+        const leftDimensions = leftEl.getBoundingClientRect();
+        const top = leftDimensions.top > 0 ? 0 : -leftDimensions.top;
+
+        const height =
+            Math.min(window.innerHeight, leftDimensions.bottom) - Math.max(leftDimensions.top, 0);
+
+        const doDisplay = leftDimensions.bottom > 0;
+
+        setRightTop(top);
+        setRightHeight(height);
+        setDoDisplayRightPane(doDisplay);
+    };
+
+    const handleResize = e => {
+        const widthEl = blockRef.current;
+        const { width: fullWidth } = widthEl.getBoundingClientRect();
+        const padding = +window
+            .getComputedStyle(widthEl, null)
+            .getPropertyValue('padding-right')
+            .match(/\d.?/)[0];
```

Comment:
Can we have a short comment what exactly is happening here? Or maybe it would be even better to move it to a separate function.